File: tests/xcalablemp/NOT-IMPLEMENTED/gmove_in_out/C/in10.c
/*testp110.c*/
/*loop指示文とpost/wait指示文のテスト*/
#include<mpi.h>
#include<xmp.h>
#include<stdio.h>
#include<stdlib.h>      
#pragma xmp nodes p(4,4)
static const int N=1000;
#pragma xmp template t(0:N-1,0:N-1)
#pragma xmp distribute t(cyclic,cyclic) onto p
int a[N][N];
int ii,jj;
#pragma xmp align a[i][j] with t(i)
int i,j;
char *result; 
int main(void){
   if(xmp_num_nodes() != 16){
      printf("%d\n","You have to run this progmam by 16 nodes.");
   }   
#pragma xmp loop (j,i) on t(j,i)
   for(i=0;i<N;i++){
      for(j=0;j<N;j++){
         a[i][j] = i;
      }
   }
   
   ii = (xmp_node_num()%4)+1;
   jj = xmp_node_num()/4+1;
#pragma xmp barrier
#pragma xmp loop (i) on t(:,i)
   for(i=2;i<N+1;i++){
      if(i != 2){
#pragma xmp wait(p(jj,(i-1)%4))
      }
#pragma xmp gmove in
      a[:][i] = a[:][i-1];
#pragma xmp loop (i) on t(j,i)
      for(i=0;i<N;i++){
         a[i][j] = a[i][j] + i*N;
      }
      if(i!=N){
#pragma xmp post(p(jj,(i+1)%4)),1
      }
   }

   result = "OK";
#pragma xmp loop (j,i) on t(j,i)
   for(i=0;i<N;i++){
      for(j=0;j<N;j++){
         if(a[i][j] != j*N+i) result = "NG";
      }
   }
   printf("%d %s %s\n",xmp_node_num(),"testp110.c",result);
   return 0;
}    
         
      
   

File: tests/XMP/NOT-IMPLEMENTED/39/in10.f90
! testp110.f
! loop指示文とpost/wait指示文のテスト

      program main
      include 'xmp_lib.h'
      integer,parameter:: N=1000
!$xmp nodes p(4,4)
!$xmp template t(N,N)
!$xmp distribute t(cyclic,cyclic) onto p
      integer a(N,N)
      integer ii, jj
!$xmp align a(i,j) with t(i,j)
      character(len=3) result

      if(xmp_num_nodes() .ne. 16) then
         print *, ' You have to run this program by 16 nodes.'
      endif
      
!$xmp loop (i,j) on t(i,j)
      do j=1, N
         do i=1, N
            a(i,j) = i
         enddo
      enddo

      ii = mod(xmp_node_num(), 4)+1
      jj = xmp_node_num()/4+1
!$xmp barrier
!$xmp loop (j) on t(:,j)
      do j=2, N
         if(j.ne.2) then
!$xmp wait(p(ii, mod(j-1,4)), 1
         endif
!$xmp gmove in
         a(:,j) = a(:,j-1)
!$xmp loop (i) on t(i,j)
         do i=1, N
            a(i,j) = a(i,j)+(j-1)*N
         enddo
         if(j.ne.N) then
!$xmp post(p(ii, mod(j+1,4)), 1
         endif
      enddo

      result = 'OK '
!$xmp loop (i,j) on t(i,j)
      do j=1, N
         do i=1, N
            if(a(i,j) .ne. (j-1)*N+i) result = 'NG '
         enddo
      enddo

      print *, xmp_node_num(), 'testp110.f ', result

      end
      
